src/Maestro/DependencyUpdateErrorProcessor/DependencyUpdateErrorProcessor.cs

```diff
@@ -0,0 +1,306 @@
+            BuildAssetRegistryContext context,
+            IOptions<DependencyUpdateErrorProcessorOptions> options
+            )
+        {
+            _stateManager = stateManager;
+            Logger = logger;
+            Context = context;
+            _options = options.Value;
+        }
+
+        private IReliableStateManager _stateManager;
+
+        private ILogger<DependencyUpdateErrorProcessor> Logger { get; }
+
+        private BuildAssetRegistryContext Context { get; }
+
+        private DependencyUpdateErrorProcessorOptions _options;
+
+        //Runs every hour in staging for now, in production it will run once a day.
+        [CronSchedule("0 0 0/1 1/1 * ? *", TimeZones.PST)]
+        public async Task DependencyUpdateErrorProcessing()
+        {
+           if (_options.ConfigurationRefresherEndPointUri != null && _options.DynamicConfigs != null)
+            {
+                await _options.ConfigurationRefresherEndPointUri.Refresh();
+                bool.TryParse(_options.DynamicConfigs["FeatureManagement:DependencyUpdateErrorProcessor"], 
+                    out var dependencyUpdateErrorProcessorFlag);
+                if (dependencyUpdateErrorProcessorFlag)
+                {
+                    IReliableDictionary<string, DateTime> update =
+                        await _stateManager.GetOrAddAsync<IReliableDictionary<string, DateTime>>("update");
+                    DateTimeOffset previousTransaction;
+                    try
+                    {
+                        using (ITransaction tx = _stateManager.CreateTransaction())
+                        {
+                            previousTransaction = await update.GetOrAddAsync(
+                             tx,
+                             "update",
+                             DateTime.UtcNow 
+                             );
+                            await tx.CommitAsync();
+                        }
+                        await CheckForErrorInRepositoryBranchHistoryTable(previousTransaction, _options.GithubUrl);
+                    }
+                    catch (Exception ex)
+                    {
+                        Logger.LogError(ex.Message, "Unable to get the previous transaction time from reliable services.");
+                    }
+                }
+            }
+            else
+            {
+                Logger.LogInformation("Dependency Update Error processor is disabled because no App Configuration was available.");
+            }
+        }
+
+        private async Task CheckForErrorInRepositoryBranchHistoryTable(DateTimeOffset previousTransaction , string issueRepo)
+        {
+            // Looking for errors from the RepositoryBranchHistory table
+            List<RepositoryBranchUpdateHistoryEntry> unprocessedHistoryEntries = (from repoBranchUpdateHistory in Context.RepositoryBranchUpdateHistory
+                                                             where repoBranchUpdateHistory.Success == false
```

Comment:
Don't indent this code so much.  Just a single indent from the starting line is sufficient.  Most of these lines are too long as a result of the large leading indent.

Reply:
okay